src/watertap_contrib/reflo/unit_models/zero_order/forward_osmosis_zo.py

```diff
@@ -0,0 +1,886 @@
+        )
+        iscale.constraint_scaling_transform(self.eq_water_mass_in_weak_draw, sf)
+
+        sf = iscale.get_scaling_factor(
+            self.weak_draw_props[0].pressure_osm_phase["Liq"]
+        )
+        iscale.constraint_scaling_transform(self.eq_weak_draw_osm_pres, sf)
+
+        sf = iscale.get_scaling_factor(self.heat_mixing)
+        iscale.constraint_scaling_transform(self.eq_heat_mixing, sf)
+
+        sf = iscale.get_scaling_factor(self.delta_temp_membrane)
+        iscale.constraint_scaling_transform(self.eq_temp_dif_membrane1, sf)
+
+        sf = iscale.get_scaling_factor(self.delta_temp_membrane)
+        iscale.constraint_scaling_transform(self.eq_temp_dif_membrane2, sf)
+
+        sf = iscale.get_scaling_factor(
+            self.product_props[0].flow_mass_phase_comp["Liq", "H2O"]
+        )
+        iscale.constraint_scaling_transform(self.eq_product_water_mass_flow, sf)
+
+        sf = iscale.get_scaling_factor(self.reg_draw_props[0].temperature)
+        iscale.constraint_scaling_transform(self.eq_reg_draw_temp, sf)
+
+        for ind, c in self.eq_reg_draw_mass_flow.items():
+            sf = iscale.get_scaling_factor(
+                self.strong_draw_props[0].flow_mass_phase_comp["Liq", ind]
+            )
+            iscale.constraint_scaling_transform(c, sf)
+
+        sf = iscale.get_scaling_factor(self.product_props[0].temperature)
+        iscale.constraint_scaling_transform(self.eq_product_temp, sf)
+
+        sf = iscale.get_scaling_factor(self.membrane_temp)
+        iscale.constraint_scaling_transform(self.eq_memb_temp, sf)
+
+    def _get_stream_table_contents(self, time_point=0):
+        return create_stream_table_dataframe(
+            {
+                "Feed Water": self.feed_props,
+                "Brine": self.brine_props,
+                "Strong Draw": self.strong_draw_props,
+                "Weak Draw": self.weak_draw_props,
+                "Regenerated Draw": self.reg_draw_props,
+                "Product Water": self.product_props,
+            },
+            time_point=time_point,
+        )
+
+    def _get_performance_contents(self, time_point=0):
+        var_dict = {}
+        var_dict["Strong draw solution volumetric flow rate (m3/s)"] = (
+            self.strong_draw_props[0].flow_vol_phase["Liq"]
+        )
+        var_dict["Weak draw solution volumetric flow rate (m3/s)"] = (
+            self.weak_draw_props[0].flow_vol_phase["Liq"]
+        )
+        var_dict["Mass fraction of weak draw solution"] = self.weak_draw_props[
+            0
+        ].mass_frac_phase_comp["Liq", "DrawSolution"]
+        var_dict["Brine salinity (g/L)"] = self.brine_props[0].conc_mass_phase_comp[
+            "Liq", "TDS"
+        ]
+        var_dict["Membrane temperature (K)"] = self.membrane_temp
+
+        return {"vars": var_dict}
+
+    @property
+    def default_costing_method(self):
+        return cost_forward_osmosis
+
+
+if __name__ == "__main__":
```

Comment:
This is interesting, we typically don't have a __main__ function in separate unit models. I am not sure if this is a good idea from a software-package-development perspective. My gut is to remove this but interested if @adam-a-a has any thoughts. I am also interested if @Zhuoran29 included this for a specific reason?

Reply:
Thanks, it's supposed to be fully removed.